httpbp/client_middlewares.go

```diff
@@ -266,28 +264,16 @@ func MaxConcurrency(maxConcurrency int64) ClientMiddleware {
 
 // MonitorClient is an HTTP client middleware that wraps HTTP requests in a
 // client span.
+//
+// This middleware always use the injected v2 tracing http client middleware.
+// If there's no v2 tracing http client middleware injected, it's no-op.
 func MonitorClient(slug string) ClientMiddleware {
 	if mw := internalv2compat.V2TracingHTTPClientMiddleware(); mw != nil {
 		return mw
 	}
 	return func(next http.RoundTripper) http.RoundTripper {
-		return roundTripperFunc(func(req *http.Request) (resp *http.Response, err error) {
-			span, ctx := opentracing.StartSpanFromContext(
-				req.Context(),
-				slug+".request",
-				tracing.SpanTypeOption{Type: tracing.SpanTypeClient},
-			)
-			span.SetTag("http.method", req.Method)
-			span.SetTag("http.url", req.URL)
-
-			defer func() {
-				span.FinishWithOptions(tracing.FinishOptions{
-					Ctx: req.Context(),
-					Err: err,
-				}.Convert())
-			}()
-			return next.RoundTrip(req.WithContext(ctx))
-		})
+		// no-op
```

Comment:
Is this always broken-broken, or could there be folks who have figured out how to get this working or for whom the semi-broken traces are still useful?

Reply:
if you define "always" as "since we retired the infra for tracing v0", then yes this is always broken :)

Reply:
the longer answer is that this worked for v0 tracing but never worked for v2, the v2 tracing is done via the injected middlewares from the compat library, and recently since we enabled interop in other places, the fallback to v0 implementation (when v2 middleware is not injected) start to spam the logs.

Reply:
Thanks.  Can we emit a warning if this case gets hit to let folks know that v0 tracing middleware is no longer functional?  I doubt anyone will see it, but it feels like the least we can do 😅

Reply:
the point of this PR is to _not_ spam the logs, if we log every time we fallback to v0, then that defeats the purpose :)